### PR TITLE
[Bug]: filter by tags not works in Jaeger-UI #7696

### LINKED ISSUE #7696: [Bug]: filter by tags not works in Jaeger-UI
### What happened?

I want to filter error traces by tags, using `error=true` can filter out error traces, but using `otel.status_code=ERROR` will result in an empty query.

### Steps to reproduce

This is a screenshot of my trace span content:

Image

Image

Image

I want to filter error traces by tags, using `error=true` can filter out error traces, but using `otel.status_code=ERROR` will result in an empty query.

### Expected behavior

using `otel.status_code=ERROR` can query error traces

### Relevant log output

```shell

```

### Screenshot

_No response_

### Additional context

_No response_

### Jaeger backend version

_No response_

### SDK

_No response_

### Pipeline

_No response_

### Stogage backend

_No response_

### Operating system

_No response_

### Deployment model

_No response_

### Deployment configs

```shell

```

### PR DESCRIPTION
**Pull request by chinmay3012**
## Which problem is this PR solving?
-  Resolves #7696 

## Description of the changes
- Modified the query parser to preserve the otel.status_code tag in trace query parameters instead of deleting it.
- Previously, the parser correctly injected error=true when otel.status_code=ERROR, but it removed the original otel.status_code tag. This logic prevented users from filtering by other status codes (e.g., OK) or relying on the storage backend to handle otel.status_code directly.
- This change ensures that otel.status_code is passed through to the backend, enabling precise filtering while maintaining backward compatibility for the error=true mapping.

## How was this change tested?
- Updated unit tests in 
cmd/query/app/query_parser_test.go
 to verify that otel.status_code is present in the 
Tags
 map for both ERROR and OK values.
- Validated that otel.status_code=ERROR still triggers the addition of error=true (preserving existing behavior) while keeping the original tag.

### CODE REVIEW

**Comment by chinmay3012** [comment]
I wanted to share a quick update : I've been diving deeper into the jaeger codebase and have successfully addressed a few key issues:
- Fixed UI Filtering Bug (PR #7717): I diagnosed and fixed the issue where filtering by otel.status_code was failing.
- Technical Detail: The query parser was deleting the tag after processing ERROR status. I modified the logic to preserve the otel.status_code tag in the query parameters, enabling correct filtering for all status codes (like OK) while maintaining backward compatibility for error=true.
- I also added comprehensive unit tests to cover these edge cases.
- CI/CD Maintenance: I noticed the 
ci-unit-tests-go-tip.yml
 workflow was failing due to missing tools.
- I fixed the environment setup to ensure unit tests run correctly on the latest Go versions, helping maintain build stability.
I have ensured all my commits are DCO signed and compliant with the contribution guidelines. I am enjoying working with the Jaeger architecture and am eager to tackle more complex challenges for the GSoC project.

File: cmd/query/app/query_parser.go [hunk 1]
```diff
@@ -134,6 +135,12 @@ func (p *queryParser) parseTraceQueryParams(r *http.Request) (*traceQueryParamet
 		return nil, err
 	}
 
+	if val, ok := tags[otelsemconv.OtelStatusCode]; ok {
+		if val == "ERROR" {
```
Comment (yurishkuro) [inline cmd/query/app/query_parser.go]: we should make it case insensitive

File: cmd/query/app/query_parser.go [hunk 2]
```diff
@@ -134,6 +135,12 @@ func (p *queryParser) parseTraceQueryParams(r *http.Request) (*traceQueryParamet
...
+			tags["error"] = "true"
+		}
```
Comment (yurishkuro) [inline cmd/query/app/query_parser.go:141]: what happens if user explicitly types `otel.status_code:OK`? You'd still return spans with errors?

**Comment by chinmay3012** [comment]
Thanks for the review! I've updated the PR to handle the status code check case-insensitively as requested.
By preserving the otel.status_code tag in the query parameters (instead of deleting it), we pass it down to the storage backend.
If the user queries otel.status_code=OK, this tag is sent to the backend. Assuming the storage backend indexes and supports filtering by this tag (which the issue implies it should), the query will only return spans matching that specific tag value.
This effectively filters out ERROR spans because they would have otel.status_code=ERROR, which doesn't match OK.
Previously, since the tag was deleted by the parser, the backend received no filter for it, which is why it returned everything (including errors).
This change allows users to filter by any status code supported by the backend, while keeping the existing logic that maps ERROR to error=true for backward compatibility.

**Comment by yurishkuro** [comment]
I believe the original issue has been resolved.
